opds_import.py

```diff
@@ -456,12 +461,20 @@ def __init__(self, _db, collection, data_source_name=None,
             self.log.warn("Metadata Wrangler integration couldn't be loaded, importing without it.")
             self.metadata_client = None
 
-        if collection and not mirror:
-            # If this Collection is configured to mirror the assets it
-            # discovers, this will create a MirrorUploader for that
-            # Collection. Otherwise, this will return None.
-            mirror = MirrorUploader.for_collection(collection)
-        self.mirror = mirror
+        if collection:
+            if not covers_mirror:
+                # If this Collection is configured to mirror the assets it
+                # discovers, this will create a MirrorUploader for that
+                # Collection for its purpose. Otherwise, this will return None.
+                covers_mirror = MirrorUploader.for_collection(
+                    collection, ExternalIntegrationLink.COVERS
+                )
+            if not books_mirror:
+                books_mirror = MirrorUploader.for_collection(
+                    collection, ExternalIntegrationLink.BOOKS
+                )
+
+        self.mirror = dict(covers=covers_mirror, books=books_mirror)
```

Comment:
Yeah, you end up with a dictionary, so take a dictionary as input and make your own if the input is `None`.